app/diary/users/get_all_users_list.py

```diff
@@ -1,37 +1,79 @@
 import json
+import os
+from decimal import Decimal
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+TABLE_NAME = "UserDiaryTable"
+PR_NUM = os.getenv("PR_NUM")
+print(f"{TABLE_NAME}-{PR_NUM}")
+dynamodb = boto3.resource("dynamodb", region_name="ap-northeast-1")
+user_diary_table = dynamodb.Table(f"{TABLE_NAME}-{PR_NUM}")
+
+
+def decimal_to_int(obj):
+    if isinstance(obj, Decimal):
+        return int(obj)
+    if isinstance(obj, set):
+        return list(obj)
 
 
 def lambda_handler(event, context):
     qsp = event.get("queryStringParameters")
     if qsp:
-        from_date = event.get("queryStringParameters", {}).get("from_date")
-        to_date = event.get("queryStringParameters", {}).get("to_date")
+        from_date = qsp.get("from_date")
+        to_date = qsp.get("to_date")
     else:
         from_date = None
         to_date = None
 
+    # バリデーション
     if from_date is not None and not isinstance(from_date, str):
         return {"statusCode": 400, "body": "Bad Request: Invalid from_date"}
     if to_date is not None and not isinstance(to_date, str):
         return {"statusCode": 400, "body": "Bad Request: Invalid to_date"}
 
-    # ここに処理を書く
-    example = [
-        {
-            "diary_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
-            "date": "2023-08-31",
-            "details": "hoge.fugaの単体テストを作成する",
-            "serious": 0,
-            "task_ids": ["3fa85f64-5717-4562-b3fc-2c963f66afa6"],
-            "created_at": "2020-01-01T00:00:00+09:00",
-            "updated_at": "2020-01-01T00:00:00+09:00",
-            "user_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
-        }
+    # TODO from user DB
+    user_list = [
+        "4f73ab32-21bf-47ef-a119-fa024bc2b9cc",
+        "595c060d-8417-4ac8-bcb5-c8e733dc64e0",
+        "e08bf311-b1bc-4a38-bac1-374c3ede7203",
     ]
 
+    # get user diary list between from_date and to_date
+    user_daily_list = []
+    for user in user_list:
+        if from_date is None and to_date is None:
+            user_id_key = Key("user_id").eq(user)
+            option = {
+                "IndexName": "UserIndex",
+                "KeyConditionExpression": user_id_key,
+            }
+        else:
+            if from_date is not None and to_date is not None:
+                datetime_range = Key("date").between(from_date, to_date)
+            elif from_date is not None:
+                datetime_range = Key("date").gte(from_date)
+            elif to_date is not None:
+                datetime_range = Key("date").lte(to_date)
+
+            user_id_key = Key("user_id").eq(user)
+            option = {
+                "IndexName": "UserDateIndex",
+                "KeyConditionExpression": user_id_key & datetime_range,
+            }
```

Comment:
こんな感じで、判定を外に出した方が幸せかも

```suggestion
    if from_date is None and to_date is None:
        option = {
            "IndexName": "UserIndex",
        }
        datetime_range = None
    else:
        if from_date is not None and to_date is not None:
            datetime_range = Key("date").between(from_date, to_date)
        elif from_date is not None:
            datetime_range = Key("date").gte(from_date)
        elif to_date is not None:
            datetime_range = Key("date").lte(to_date)
        option = {
            "IndexName": "UserDateIndex",
        }

    user_daily_list = []
    for user in user_list:
            user_id_key = Key("user_id").eq(user)
            if datetime_range is not None:
                expr = user_id_key & datetime_range
            else:
                expr = user_id_key
            option["KeyConditionExpression"] = expr
```